src/main/java/com/googleintern/wfm/ruleengine/action/service/RuleSuggestionServiceImplementation.java

```diff
@@ -59,21 +58,38 @@ public static void main(String[] args) throws IOException, CsvException {
   public String suggestRules(String csvFilePath) throws IOException, CsvException {
     ImmutableList<UserModel> userPoolAssignments = CsvParser.readFromCSVFile(csvFilePath);
 
-    ImmutableList<UserModel> validUserPoolAssignments =
-        DataProcessor.filterValidData(userPoolAssignments);
+    ImmutableList<UserModel> usersWithValidWorkgroupId =
+        DataProcessor.filterUsersWithValidWorkgroupId(userPoolAssignments);
 
-    ImmutableSet<RuleModel> rules = suggestRules(validUserPoolAssignments);
+    ImmutableSet<RuleModel> rules = suggestRules(usersWithValidWorkgroupId);
 
     CsvWriter.writeDataIntoCsvFile(CSV_OUTPUT_FILE_PATH, rules.asList());
 
-    RuleValidation ruleValidation = new RuleValidation(validUserPoolAssignments);
+    RuleValidation ruleValidation = new RuleValidation(usersWithValidWorkgroupId);
     ruleValidation.validate(rules).writeToCsvFile(CSV_VALIDATION_RESULT_FILE_PATH);
     return null;
   }
 
   @Override
-  public String suggestRules(String csvFilePath, int percentage) {
-    throw new NotImplementedException();
+  public String suggestRules(String csvFilePath, boolean assignMorePermissions)
+      throws IOException, CsvException {
+    ImmutableList<UserModel> userPoolAssignments = CsvParser.readFromCSVFile(csvFilePath);
+
+    ImmutableList<UserModel> usersWithValidWorkgroupId =
+        DataProcessor.filterUsersWithValidWorkgroupId(userPoolAssignments);
+
+    ImmutableList<UserModel> validUsers =
+        assignMorePermissions == true
```

Comment:
you don't need to compare assignMorePermissions as it is already boolean

```suggestion
        assignMorePermissions
```